### ISSUE TITLE
Doesn't work on Surface Pro X

### ISSUE THREAD

**Issue by tommyvct** [aarch64-laptops/debian-cdimage#6]
Will inventing a new device tree `.dts` file enough to get it working?
![image](https://user-images.githubusercontent.com/7903172/105624591-92b40800-5de8-11eb-81d9-8e2113fd0e79.png)

**Comment by shawnguo2** [comment]
No.  I believe Surface Pro X uses a different SoC from Yoga C630 one.  I'm about to receive Lenovo Flex 5G, which should use a same (almost?) SoC as Surface Pro X.  After I get the CD image work on Flex 5G, you then stand a better chance to have the installer work on the Surface.

**Comment by tommyvct** [comment]
Hi, I tried the latest 0.4 image. It now boots up! Thanks for your work!

But the keyboard cover only works in `grub`, once we get in Linux the touchpad and the keyboard doesn't work at all. I tried a pair of USB keyboard and mouse and it worked. 

There is a kernel panic happening on reboot, too:
![image](https://user-images.githubusercontent.com/7903172/111385651-1ec00e80-8679-11eb-9782-6e321d5009fa.png)

**Comment by qzed** [comment]
> But the keyboard cover only works in `grub`, once we get in Linux the touchpad and the keyboard doesn't work at all.

The device uses a somewhat special method of handling input: Similar to the Surface Laptops and Surface Book 3, the Surface Pro X routs its HID input devices over the Surface Aggregator Module (SAM). SAM is connected to the SoC via a UART and the main issue right now is getting the UART driver to work.

See https://github.com/Sonicadvance1/linux/issues/27 for an all-over-the-place discussion on the Surface Pro X and what's needed to get it working with Linux.
